solutions/swb-reference/src/services/swbProjectService.ts

```diff
@@ -0,0 +1,287 @@
+      projectId,
+      researcherRole
+    );
+
+    const createIdentityPermissionsRequest = CreateIdentityPermissionsRequestParser.parse({
+      authenticatedUser: user,
+      identityPermissions: [...paIdentityPermissions, ...researcherIdentityPermissions]
+    });
+
+    await this._dynamicAuthorizationService.createIdentityPermissions(createIdentityPermissionsRequest);
+
+    return createProjectResponse;
+  }
+
+  /**
+   * Get project
+   * @param request - the request object for getting a project
+   *
+   * @returns Project entry in DDB
+   */
+  public async getProject(request: GetProjectRequest): Promise<Project> {
+    return this._projectService.getProject(request);
+  }
+
+  /**
+   * Get projects
+   * @param request - the request object for getting a project
+   *
+   * @returns Project entry in DDB
+   */
+  public async getProjects(request: GetProjectsRequest): Promise<Project[]> {
+    return this._projectService.getProjects(request);
+  }
+
+  /**
+   * List projects
+   *
+   * @param request - the request object for listing projects
+   * @returns Project entries in DDB, with optional pagination token
+   */
+  public async listProjects(request: ListProjectsRequest): Promise<PaginatedResponse<Project>> {
+    const getUserGroupsResponse = await this._dynamicAuthorizationService.getUserGroups({
+      authenticatedUser: request.user,
+      userId: request.user.id
+    });
+    return this._projectService.listProjects(request, getUserGroupsResponse.data.groupIds);
+  }
+
+  /**
+   * Soft deletes a project from the database.
+   *
+   * @param request - a {@link DeleteProjectRequest} object that contains the id of the project to delete
+   * @param checkDependencies - an async function that checks if there are dependencies associated with the project
+   */
+  public async softDeleteProject(
+    request: DeleteProjectRequest,
+    checkDependencies: (projectId: string) => Promise<void>
+  ): Promise<void> {
+    await this._projectService.softDeleteProject(request, checkDependencies);
+
+    const authenticatedUser = request.authenticatedUser;
+    const projectId: string = request.projectId;
+    const paRole: string = getProjectAdminRole(projectId);
+    const researcherRole: string = getResearcherRole(projectId);
+
+    // Get PA Permissions to delete
+    const paIdentityPermissions: IdentityPermission[] = this._generatePAIdentityPermissions(
+      projectId,
+      paRole
+    );
+
+    // Get Researcher Permissions to delete
+    const researcherIdentityPermissions: IdentityPermission[] = this._generateResearcherIdentityPermissions(
+      projectId,
+      researcherRole
+    );
+
+    const deleteIdentityPermissionsRequest = DeleteIdentityPermissionsRequestParser.parse({
+      authenticatedUser,
+      identityPermissions: [...paIdentityPermissions, ...researcherIdentityPermissions]
+    });
+
+    await this._dynamicAuthorizationService.deleteIdentityPermissions(deleteIdentityPermissionsRequest);
```

Comment:
There are some helpers in dynamic auth which may help you here. If you are just deleting everything for the subject, consider `deleteSubjectIdentityPermissions`.
There are also a couple of bulk 'get' operations that may help you enumerate what you need to delete:
`getIdentityPermissionsBySubject` and `getIdentityPermissionsByIdentity`

Reply:
I'll look into using deleteSubjectIdentityPermissions; will get input from @htztomic on if we could actually get by with just deleteGroup.

Reply:
`this._dynamicAuthorizationService.deleteGroup()` will delete the permissions associated with the group